CryptoPkg/Library/BaseCryptLib/SysCall/CrtWrapper.c

```diff
@@ -271,7 +271,46 @@ strcpy (
   const char  *strSource
   )
 {
-  AsciiStrCpyS (strDest, MAX_STRING_SIZE, strSource);
+  AsciiStrCpyS (strDest, AsciiStrnSizeS (strSource, MAX_STRING_SIZE), strSource);
+  return strDest;
+}
+
+char *
+strncpy (
+  char        *strDest,
+  const char  *strSource,
+  size_t      count
+  )
+{
+  UINTN  DestMax = MAX_STRING_SIZE;
+
+  if (count < MAX_STRING_SIZE) {
+    DestMax = count + 1;
+  } else {
+    count = MAX_STRING_SIZE-1;

```

Comment:
How about changing the code to:
```
  } else {
    count = MAX_STRING_SIZE;
```
Because in the `AsciiStrnCpyS `description:
```
  Copies not more than Length successive char from the string pointed to by
  Source to the array pointed to by Destination. If no null char is copied from
  Source, then Destination[Length] is always set to null.

  This function is similar as strncpy_s defined in C11.

  If an error is returned, then the Destination is unmodified.

  @param  Destination              A pointer to a Null-terminated Ascii string.
  @param  DestMax                  The maximum number of Destination Ascii
                                   char, including terminating null char.
  @param  Source                   A pointer to a Null-terminated Ascii string.
  @param  Length                   The maximum number of Ascii characters to copy.
```

Reply:
I interpreted MAX_STRING_SIZE as the maximum size including the Null character.
`@param  DestMax                  The maximum number of Destination Ascii
                                   char, including terminating null char.`

If it takes this branch it would call:

`AsciiStrnCpyS (Destination, MAX_STRING_SIZE, Source, MAX_STRING_SIZE-1)` because a Length of MAX_STRING_SIZE would not fit.

And in SafeString.c:1907:
`
  if (Length >= DestMax) {
    SAFE_STRING_CONSTRAINT_CHECK ((DestMax > SourceLen), RETURN_BUFFER_TOO_SMALL);
  }
`
it would abort.

Reply:
I got it. Thanks.